sdk/nodejs/storage/zMixins.ts

```diff
@@ -401,27 +419,68 @@ export class QueueEventSubscription extends appservice.EventSubscription<QueueCo
     }
 }
 
+function merge(trigger: appservice.BindingDefinition, appSettings: pulumi.Input<{ [key: string]: string }>, otherBindings?: appservice.BindingSettings[]) {
+    if (otherBindings) {
+        const bindings = pulumi.all(otherBindings.map(b => b.binding)).apply(bs => [trigger, ...bs]);
+
+        const perFunctionSettings = otherBindings.map(b => b.settings);
+        const settings =  pulumi.all([appSettings, ...perFunctionSettings]).apply(items => items.reduce((a, b) => ({ ...a, ...b }), {}));
+        return { bindings, settings };
+    } else
+    {
+        return { bindings: [trigger], settings: appSettings };
+    }
+}
+
 /**
  * Azure Function triggered by a Storage Queue.
  */
-export class QueueFunction extends appservice.FunctionBase<QueueContext, Buffer, void> {
+export class QueueFunction extends appservice.FunctionBase<QueueContext, Buffer, void | Record<string, any>> {
     constructor(name: string, args: QueueFunctionArgs) {
         const bindingConnectionKey = pulumi.interpolate`${args.queue.storageAccountName}ConnectionStringKey`;
 
         const account = pulumi.all([args.queue.resourceGroupName, args.queue.storageAccountName])
             .apply(([resourceGroupName, storageAccountName]) =>
                 storage.getAccount({ resourceGroupName, name: storageAccountName }));
 
-        const appSettings = pulumi.all([account.primaryConnectionString, bindingConnectionKey]).apply(
+        const connectionStingsSetting = pulumi.all([account.primaryConnectionString, bindingConnectionKey]).apply(
             ([connectionString, key]) => ({ [key]: connectionString }));
 
-        super(name, [<QueueBindingDefinition>{
+        const trigger: QueueBindingDefinition = {
             name: "queue",
             type: "queueTrigger",
             direction: "in",
             dataType: "binary",
             queueName: args.queue.name,
             connection: bindingConnectionKey,
-        }], args, appSettings);
+        };
+
+        const { bindings, settings } = merge(trigger, connectionStingsSetting, args.bindings);
+
+        super(name, bindings, args, settings);
+    }
+}
+
+export class QueueOutputBinding implements appservice.BindingSettings {
+    public readonly binding: pulumi.Input<appservice.BindingDefinition>;
+    public readonly settings: pulumi.Input<{ [key: string]: any; }>;
+
+    constructor(name: string, queue: Queue) {
+        const bindingConnectionKey = pulumi.interpolate`${queue.storageAccountName}ConnectionStringKey`;

```

Comment:
This is repeated to the connection string building in QueueFunction - is there a way to make sure they really use the same connection string?

Reply:
I combined both into `resolveAccount` function. Is that what you mean?

Reply:
It's even better than what I meant - i had meant just the name of the connection string.